classes/helper.php

```diff
@@ -0,0 +1,258 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_crawler;
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Provide helper functions for crawling on a course
+ *
+ * @package tool_crawler
+ * @author  Nathan Nguyen <nathannguyen@catalyst-au.net>
+ * @copyright  Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class helper {
+    // Reference: https://datatracker.ietf.org/doc/html/rfc7231.
+
+    /** @var $httpcodes list of http code */
+    private static $httpcodes = [
```

Comment:
I do not think this map should exist, if we show an http code then we should show the http string that the server returns and not make one up based on the code alone.

Reply:
Client wants a layman explanation of httpcode so I added some brief info

Reply:
Wow! Ok today I learnt something major. In http1.1 you would get back a head something like:

HTTP/1.1 404 Not Found

The Code is '404' and the Reason-Phrase is 'Not Found' and we could have captured this and used it directly instead of making it up in this mapping. In http2 this is dropped completely and there is no reason phrase at all which is really surprising and will have impacts in lots of places.

So yes I guess we do need this mapping. In which case can you make these language strings

Reply:
yep sure, I will change them as lang strings

Reply:
Hi @brendanheywood 

I have changed them to lang strings